packages/tizen_rpc_port/example/example_proxy/lib/message_proxy.dart

```diff
@@ -0,0 +1,262 @@
+
+  /// This abstract method will be called when the delegate is received event.
+  Future<void> onReceived(String sender, String msg);
+
+  @override
+  Future<void> _onReceivedEvent(Parcel parcel) async {
+    final String sender = parcel.readString();
+    final String msg = parcel.readString();
+
+    await onReceived(sender, msg);
+  }
+}
+
+/// Abstract class for creating [Message] class for RPC.
+abstract class Message extends ProxyBase {
+  /// Constructor for this class.
+  Message(String appid) : super(appid, 'Message');
+
+  bool _online = false;
+  final List<CallbackBase> _delegateList = <CallbackBase>[];
+
+  /// This abstract method will be called when the connet() is succeed.
+  Future<void> onConnected();
+
+  /// This abstract method will be called when the connection with the stub is disconnected.
+  Future<void> onDisconnected();
+
+  /// This abstract method will be called when connect() is failed.
+  Future<void> onRejected(String errorMessage);
+
+  @override
+  @nonVirtual
+  Future<void> onConnectedEvent() async {
+    _online = true;
+    await onConnected();
+  }
+
+  @override
+  @nonVirtual
+  Future<void> onDisconnectedEvent() async {
+    _online = false;
+    await onDisconnected();
+  }
+
+  @override
+  @nonVirtual
+  Future<void> onRejectedEvent(String errorMessage) async {
+    await onRejected(errorMessage);
+  }
+
+  @override
+  @nonVirtual
+  Future<void> onReceivedEvent(Parcel parcel) async {
+    final int cmd = parcel.readInt32();
+    if (cmd != _MethodId.callback.id) {
+      return;
+    }
+
+    await _processReceivedEvent(parcel);
+  }
+
+  Future<void> _processReceivedEvent(Parcel parcel) async {
+    final int id = parcel.readInt32();
+    final int seqId = parcel.readInt32();
+    final bool once = parcel.readBool();
+
+    for (final CallbackBase delegate in _delegateList) {
+      if (delegate._id == id && delegate._seqId == seqId) {
+        await delegate._onReceivedEvent(parcel);
+        if (delegate._once && once) {
+          _delegateList.remove(delegate);
+        }
+        break;
+      }
+    }
+  }
+
+  Future<Parcel> _consumeCommand(Port port) async {
+    while (true) {
+      try {
+        final Parcel parcel = Parcel.fromPort(port);
+        final int cmd = parcel.readInt32();
+        if (cmd == _MethodId.result.id) {
+          return parcel;
+        }
+      } catch (e) {
+        Log.error(_logTag, e.toString());
+        return Parcel();
+      }
+    }
+  }
+
+  /// Connects with the stub application.
+  @override
+  Future<void> connect() async {
+    Log.info(_logTag, 'connect()');
+    await super.connect();
+  }
+
+  /// Dispose registered delegate interface.
+  void disposeCallback(String tag) {
+    Log.info(_logTag, 'disposeCallback($tag)');
+    _delegateList.removeWhere((CallbackBase element) => element.tag == tag);
+  }
+
+  /// This method is used to send 'Register' request to the stub app.
+  Future<int> register(String name, NotifyCB cb) async {
```

Comment:
Apparently there's no async operation in this method but do you want this method to be async in case the implementation is changed in the future?

Reply:
Yes. We've discussed it before, and we think it's better to leave it `async`.

Reply:
Okay, then you could at least make `_consumeCommand` synchronous.